frontend/src/queries/nodes/DataTable/DataTableOpenEditor.tsx

```diff
@@ -0,0 +1,56 @@
+import { LemonButton } from '@posthog/lemon-ui'
+import { useValues } from 'kea'
+import { IconTableChart } from 'lib/lemon-ui/icons'
+import { LemonMenuOverlay } from 'lib/lemon-ui/LemonMenu/LemonMenu'
+import { urls } from 'scenes/urls'
+
+import { DataTableNode, NodeKind } from '~/queries/schema'
+
+import { dataTableLogic } from './dataTableLogic'
+
+interface DataTableOpenEditorProps {
+    query: DataTableNode
+    setQuery?: (query: DataTableNode) => void
+}
+
+export function DataTableOpenEditor({ query }: DataTableOpenEditorProps): JSX.Element | null {
+    const { response } = useValues(dataTableLogic)
+
+    return (
+        <LemonButton
+            type="secondary"
+            icon={<IconTableChart />}
+            to={urls.insightNew(undefined, undefined, JSON.stringify(query))}
+            sideAction={
+                response?.hogql
+                    ? {
+                          dropdown: {
+                              overlay: (
+                                  <LemonMenuOverlay
+                                      items={[
+                                          {
+                                              label: 'Open as direct SQL insight',
```

Comment:
I find the wording not ideal. What is a `direct SQL insight`? Or do we mean `Open directly as SQL insight?` or simply `Open SQL insight`?